### ISSUE TITLE
DecloakToFire=true cannot work for cloak vehicles since phobos b32-v0.3 RC2

### ISSUE THREAD

**Issue by reedom114514** [Phobos-developers/Phobos#1109]
### Description

From my friend's link: https://www.bilibili.com/opus/820342056389967908?spm_id_from=444.41.0.0
Key point translated in English: Phobos changes the deploy fire, in vanilla YR/ares, decloaktofire=true on vehicles makes vehicles need to decloak then deploy to fire, but after B32RC2, decloaktofire=true is broken, the only way is to set decloaktofire=false and using some other ways to force the vehicle decloak, and sometimes it even cannot work either.

### Phobos Version

B36

### Conditions to reproduce

Normal YR

### INI code

```ini
[SomeVehicle]
Cloakable=yes
Secondary=A weapon which needs decloak to fire
```

### Steps to reproduce

1. producing the vehicle.
2. deploy.

### Expected behaviour

Unit deployed and decloak.

### Actual behaviour

It cannot fire.

### Checklist

- [X] The issue happens on the **latest official** version of Phobos and wasn't fixed yet.
- [X] I agree to elaborate the details if requested and provide thorough testing if the bugfix is implemented.
- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search and read the issue list to find a similar issue and didn't find it.
- [X] I have attached as much information as possible *(screenshots, gifs, videos, debug and exception logs, etc)*.

**Comment by Starkku** [comment]
Should be fixed in bca32b32947ffcbd05189fece693cc4937899a46. Please test and report back.

**Comment by reedom114514** [comment]
Okay, it is done! Thanks a lot!
